cranelift/codegen/src/isa/aarch64/inst.isle

```diff
@@ -1546,6 +1559,51 @@
             (_ Unit (emit (MInst.LoadAcquire ty dst addr))))
         (writable_reg_to_reg dst)))
 
+(type SetsFlags (enum (Inst (inst MInst))))
```

Comment:
Interesting approach here! I kind of like the functional design (flags consumer just takes a flags producer) vs. what we have in x64 (separate toplevel combinator takes a consumer and producer).

That said, this abstraction won't allow us to reuse flags later, unless I'm missing some approach -- e.g. we couldn't have two separate `csel`s, or a `csel` and an `adc`, or something like that, both use one compare. That was part of the reasoning for the top-level combinator approach. I'd also sort of prefer to be consistent if we can, just so folks jumping back and forth between the machine backends don't have to shift mindset more than necessary (admittedly this purely aspirational and we have plenty of differences in the handwritten code).

Was there a reason you had in mind to prefer this approach over the x64-style one, or just preference? If not, and if it's not too much trouble (and sorry for pushing a bit more on this!), I think it might be good to take the x64-style approach... thanks!

Reply:
Oh when I originally looked at `ProducesFlags` it didn't seem right because it has a `Reg` payload and I thought that `tst` didn't produce any registers so it was intened for something else. After reading how this is used with x64 though I see that the field can be `(invalid_reg)` which means I was able to switch over to the existing functions in `prelude.isle` for this.